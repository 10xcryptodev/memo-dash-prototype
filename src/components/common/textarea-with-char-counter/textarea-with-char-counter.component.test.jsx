import React from 'react'
import { shallow } from 'enzyme'

import TextAreaWithCharCounter from './textarea-with-char-counter.component'

describe('<TextAreaWithCharCounter />', () => {
  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('should render', () => {
    it('without crashing', () => {
      const wrapper = shallow(<TextAreaWithCharCounter />)
      expect(wrapper).toMatchSnapshot()
    })

    it('with max length', () => {
      const wrapper = shallow(<TextAreaWithCharCounter maxLength="10" />)
      expect(wrapper).toMatchSnapshot()
    })

    it('with content', () => {
      const wrapper = shallow(<TextAreaWithCharCounter value="value" maxLength="10" />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
