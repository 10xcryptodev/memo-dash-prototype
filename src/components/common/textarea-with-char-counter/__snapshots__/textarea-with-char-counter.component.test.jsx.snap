// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaWithCharCounter /> should render with content 1`] = `
<div
  className="text-area-with-char-counter"
>
  <Label
    attached="bottom right"
  >
    5
    /
    10
  </Label>
  <FormTextArea
    as={[Function]}
    control={[Function]}
    maxLength="10"
    value="value"
  />
</div>
`;

exports[`<TextAreaWithCharCounter /> should render with max length 1`] = `
<div
  className="text-area-with-char-counter"
>
  <Label
    attached="bottom right"
  >
    0
    /
    10
  </Label>
  <FormTextArea
    as={[Function]}
    control={[Function]}
    maxLength="10"
  />
</div>
`;

exports[`<TextAreaWithCharCounter /> should render without crashing 1`] = `
<div
  className="text-area-with-char-counter"
>
  <FormTextArea
    as={[Function]}
    control={[Function]}
  />
</div>
`;
